source/Weather.h

```diff
@@ -50,7 +51,7 @@ class Weather {
 	// The origin of the hazard.
 	const Point &Origin() const;
 	// Create any environmental effects and decrease the lifetime of this weather.
-	void Step(std::vector<Visual> &newVisuals);
+	void Step(std::vector<Visual> &newVisuals, const Ship *flagship = nullptr);
```

Comment:
```suggestion
	void Step(std::vector<Visual> &newVisuals, const Point *effectCenter = nullptr);
```
You don't need an _instance of the Ship object class_ for what you are trying to do, nor should the Weather class ever need to know about the Ship class. All you need is a position.